MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1365,163 +1382,133 @@ std::string EnggDiffractionViewQtGUI::fittingPeaksData() const {
-    }
+    if (!m_fitting_runno_dir_vec.empty()) {
 
-    try {
-      // add bank to the combo-box and list view
-      addBankItems(splitBaseName, focusedFile);
-    } catch (std::runtime_error &re) {
-      userWarning("Unable to insert items: ",
-                  "Could not add banks to "
-                  "combo-box or list widget; " +
-                      static_cast<std::string>(re.what()) +
-                      ". Please try again");
-    }
-  } catch (std::runtime_error &re) {
-    userWarning("Invalid file", "Unable to select the file; " +
-                                    static_cast<std::string>(re.what()));
-    return;
-  }
-}
+      // delete previous bank added to the list
+      m_uiTabFitting.comboBox_bank->clear();
 
-void EnggDiffractionViewQtGUI::updateFittingDirVec(std::string &bankDir,
-                                                   std::string &focusedFile) {
+      for (size_t i = 0; i < m_fitting_runno_dir_vec.size(); i++) {
+        Poco::Path vecFile(m_fitting_runno_dir_vec[i]);
+        std::string strVecFile = vecFile.toString();
+        // split the directory from m_fitting_runno_dir_vec
+        std::vector<std::string> vecFileSplit =
+            splitFittingDirectory(strVecFile);
 
-  try {
+        // get the last split in vector which will be bank
+        std::string bankID = (vecFileSplit[vecFileSplit.size() - 1]);
+
+        bool digit = isDigit(bankID);
 
-    std::string cwd(bankDir);
-    Poco::DirectoryIterator it(cwd);
-    Poco::DirectoryIterator end;
-    while (it != end) {
-      if (it->isFile()) {
-        std::string itFilePath = it->path();
-        Poco::Path itBankfPath(itFilePath);
-
-        std::string itbankFileName = itBankfPath.getBaseName();
-        // check if it not any other file.. e.g: texture
-        if (itbankFileName.find(focusedFile) != std::string::npos) {
-          m_fitting_runno_dir_vec.push_back(itFilePath);
+        if (digit) {
+          m_uiTabFitting.comboBox_bank->addItem(QString::fromStdString(bankID));
+
+        } else {
+          m_uiTabFitting.comboBox_bank->addItem(QString("Bank %1").arg(i + 1));
         }
       }
-      ++it;
+
+      m_uiTabFitting.comboBox_bank->setEnabled(true);
+    } else {
+      // upon invalid file
+      // disable the widgets when only one related file found
+      m_uiTabFitting.comboBox_bank->setEnabled(false);
+
+      m_uiTabFitting.comboBox_bank->clear();
     }
+
+    setDefaultBank(splittedBaseName, selectedFile);
+
   } catch (std::runtime_error &re) {
-    userWarning("Invalid file", "File not found in the following directory; " +
-                                    bankDir + ". " +
-                                    static_cast<std::string>(re.what()));
+    userWarning("Unable to insert items: ",
+                "Could not add banks to "
+                "combo-box or list widget; " +
+                    static_cast<std::string>(re.what()) + ". Please try again");
```

Comment:
This is also thinking too much for the view, which should be mostly passive. It is processing the vectors, checking the values, deciding if and what error message to give. That the view calls `userWarning()` itself is not a good sign. That kind of action should be done by the presenter: `m_view->userWaning(...)`. Then you can have the `EXPECT_CALL`s and similar in the tests. Otherwise this remains untestable and makes the view more complex and prone to develop issues.